spec/features/merchants/items/index_spec.rb

```diff
@@ -54,8 +73,58 @@
   scenario 'visitor sees the name of all items of particular merchant as links' do
     expect(current_path).to eq(merchant_items_path(@merchant_1.id))
 
-    expect(page).to have_link("#{@item_1.name}", href: merchant_item_path(@merchant_1.id, @item_1.id))
-    expect(page).to have_link("#{@item_2.name}", href: merchant_item_path(@merchant_1.id, @item_2.id))
-    expect(page).to have_link("#{@item_3.name}", href: merchant_item_path(@merchant_1.id, @item_3.id))
+    within "#all_items-#{@merchant_1.id}" do
+      expect(page).to have_link("#{@item_1.name}", href: merchant_item_path(@merchant_1.id, @item_1.id))
+      expect(page).to have_link("#{@item_2.name}", href: merchant_item_path(@merchant_1.id, @item_2.id))
+      expect(page).to have_link("#{@item_3.name}", href: merchant_item_path(@merchant_1.id, @item_3.id))
+      expect(page).to have_link("#{@item_4.name}", href: merchant_item_path(@merchant_1.id, @item_4.id))
+      expect(page).to have_link("#{@item_5.name}", href: merchant_item_path(@merchant_1.id, @item_5.id))
+      expect(page).to have_link("#{@item_6.name}", href: merchant_item_path(@merchant_1.id, @item_6.id))
+      expect(page).to have_link("#{@item_7.name}", href: merchant_item_path(@merchant_1.id, @item_7.id))
+      expect(page).to have_link("#{@item_8.name}", href: merchant_item_path(@merchant_1.id, @item_8.id))
+    end
+  end
+
+  scenario 'visitor sees the names of the top 5 most popular items ranked by total revenue generated' do
+    expect(page).to have_content('Top 5 Items')
+    first = find("#item#{@item_7.id}")
+    second = find("#item#{@item_1.id}")
+    third = find("#item#{@item_3.id}")
+    fourth = find("#item#{@item_8.id}")
+    fifth = find("#item#{@item_2.id}")
+    expect(first).to appear_before(second)
+    expect(second).to appear_before(third)
+    expect(third).to appear_before(fourth)
+    expect(fourth).to appear_before(fifth)
+  end
+
+  scenario 'visitor sees that each item name in the top item list also links to that item show page' do
+    within "#top_five_items-#{@merchant_1.id}" do
+      expect(page).to have_link(@item_7.name)
+      expect(page).to have_link(@item_1.name)
+      expect(page).to have_link(@item_3.name)
+      expect(page).to have_link(@item_8.name)
+      expect(page).to have_link(@item_2.name)
+    end
+  end
+
+  scenario 'visitor sees the total revenue generated next to each item name' do
+    within "#top_five_items-#{@merchant_1.id}" do
+      expect(page).to have_content(@merchant_1.top_five_items[0].total_revenue)
+      expect(page).to have_content(@merchant_1.top_five_items[1].total_revenue)
+      expect(page).to have_content(@merchant_1.top_five_items[2].total_revenue)
+      expect(page).to have_content(@merchant_1.top_five_items[3].total_revenue)
+      expect(page).to have_content(@merchant_1.top_five_items[4].total_revenue)
+    end
+  end
+
```

Comment:
Nice use of within blocks to scope your content seaarches.

Reply:
Thank you!!!